apps/prairielearn/src/pages/instructorCopyPublicCourseInstance/instructorCopyPublicCourseInstance.ts

```diff
@@ -0,0 +1,33 @@
+import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
+
+import * as error from '@prairielearn/error';
+
+import { copyCourseInstanceBetweenCourses } from '../../lib/copy-content.js';
+import { selectCourseInstanceById } from '../../models/course-instances.js';
+import { selectCourseById } from '../../models/course.js';
+
+const router = Router();
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const courseInstance = await selectCourseInstanceById(req.body.course_instance_id);
+    if (courseInstance === null) {
+      throw new error.HttpStatusError(404, 'Not Found');
+    }
```

Comment:
Nit: since this function can return null, it should be named `selectOptionalCourseInstanceById`. While you're here, consider refactoring that, possibly in another PR first if that'd impact a lot of the codebase.

Reply:
done: https://github.com/PrairieLearn/PrairieLearn/pull/12142